hardhat.config.js

```diff
@@ -10,14 +10,15 @@ module.exports = {
     solidity: "0.8.2",
     networks: {
         hardhat: {},
-        mumbai: {
-            url: MUMBAI_API_URL,
-            accounts: [`0x${PRIVATE_KEY}`],
-        },
-        polygon: {
-            url: POLYGON_API_URL,
-            accounts: [`0x${PRIVATE_KEY}`],
-        },
+        // Uncomment according to the network you wanna use
```

Comment:
Hmmm, why do we need to comment out these lines? Is it because of the missing `.env` file? If so, maybe we can have a default one generated in the container like we do in the CI (see [here](https://github.com/underfitted/bnomial-web3/blob/7833cb76c7295f8ff524c16c761953a82a180aa3/.github/workflows/test.yml#L31))? Or maybe another way to work around this in the config file

Reply:
The way how it was if someone tried to run locally using the Hardhat dummy network (which demands no configuration at all) it would be necessary to set a fake private key (`npm run build` complains about values undefined and formats). It felt like old practices we had before things like [12-factor app](https://12factor.net). In fact, commenting them out isn't a good solution either.

If we want to use the ideas from 12-factor apps, we should change the configuration file for something like this:

````
...
networks: {
      hardhat: {},
      deploynet: {
          url: https://myaccessprovider.com/api/v2/${API_KEY},
          accounts: [${PRIVATE_KEY}],
      },
},
etherscan: {
      apiKey: ${POLYSCAN_KEY},
},
...
````

We would avoid grouping configuration by deployment perimeter and it would allow us to use the same deployment asset (the web3 package, a docker image, etc) and promote it through a pipeline without touching the config file (e.g. the same container image could run in Mumbai, Polygon or a new private network we want just by changing the container ENV variables).

Reply:
I guess that these changes wouldn't fit well in this branch. We should keep the way it was before my commits and discuss the configuration file (if possible) in another branch.

Reply:
I'll look into that today. There should be a way to set some defaults so to not block people without `.env` setup.

Reply:
I like the idea of refactoring the config file, but as you say let's do it on another branch. In the meantime I think I have a nice workaround :)